misc-utils/waitpid.c

```diff
@@ -0,0 +1,226 @@
+/*
+ * waitpid(1) - wait for process termination
+ *
+ * Copyright (C) 2022 Thomas Weißschuh <thomas@t-8ch.de>
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it would be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+
+#include <sys/epoll.h>
+#include <assert.h>
+#include <stdlib.h>
+#include <errno.h>
+#include <string.h>
+#include <unistd.h>
+#include <stdbool.h>
+#include <getopt.h>
+
+#include "pidfd-utils.h"
+#include "c.h"
+#include "nls.h"
+
+#define EXIT_NOT_IMPLEMENTED 2
+
+static bool verbose = false;
+
+static pid_t *parse_pids(size_t n_strings, char * const *strings)
+{
+	pid_t *pids = calloc(n_strings, sizeof(*pids));
+	if (!pids)
+		return NULL;
+
+	for (size_t i = 0; i < n_strings; i++) {
+		char *endptr;
+		const char *arg = strings[i];
+		long long pid = strtoll(arg, &endptr, 10);
+		if (!endptr || pid <= 0) {
+			warnx("invalid PID: '%s'", arg);
```

Comment:
I guess  `pid_t pid = strtoul_or_err(strings[i], _("failed to parse pid"));` would be enough. Or maybe we can add strtopid_or_err() to include/strutils.h.